Makefile.rules

```diff
@@ -412,6 +412,10 @@ gap$(EXEEXT): $(OBJS) cnf/GAP-LDFLAGS cnf/GAP-LIBS cnf/GAP-OBJS
 
 endif
 
+ifeq ($(BUILD_LIBGAP),yes)
+libgap.so: $(OBJS)
```

Comment:
This is problematic. First off, `.so` is not quite portable; e.g. on macOS, it should be `.dylib`, and on Windows `.dll`

Anyway, we already *do* generated a shared library (see `libgap.la`) using libtool, in a fully portable fashion, including setting rpaths etc. -- so can't we just use that? I guess I don't quite know how you "use" libgap later on, so it's a bit difficult for me to evaluate.